### ISSUE TITLE
[BUG] Firefox intercepted `response.body()` is not consistent, potentially making repeat requests.

### ISSUE THREAD

**Issue by michaeleveringham** [microsoft/playwright-python#1902]
**Context:**
- Playwright Version: 1.27.1
- Operating System: Mac
- Python version: 3.10.3
- Browser: Firefox (compared to Chromium)
- Extra: Using a socks proxy, likely not related though.

**Code Snippet**

```python
from playwright.sync_api import sync_playwright

def handle_response(response):
    print(response.body())
    return response

playwright = sync_playwright().start()
browser = playwright.firefox.launch(
    headless=False,
    args=[
        "--disable-blink-features",
        "--disable-blink-features=AutomationControlled",
        "--no-sandbox",
    ],
    proxy="socks5://<host>:<port>",
)
context = browser.new_context()
page = context.new_page()
context.on("response", handle_response)
page.goto("https://site.url")
```

**Bug Description**

I am unsure if this is truly a bug or intentional behaviour I couldn't find documented. 

Background: I am working with a finnicky API that will return empty responses with `200` status codes when requests are repeated or malformed (truthfully, I haven't figured out exactly why, I just know that if I try to recreate the request via `httpx` or `requests` this happens). When manually browsing the page in my browser, I can see the XHR requests' response bodies with the data I want.

Issue: When using Firefox to intercept these requests, the body will often be empty when calling `response.body()` (thus raising an error on `response.json()`, despite the same request body having content in the dev tools network pane. Using Chromium, this error does not occur. I noticed that when debugging and using Firefox, calling `response.body()` is slow to resolve (~2-4 seconds), which indicates to me that perhaps in Firefox this method is repeating calls to resolve this method? If so, this should probably be documented somewhere as a call repeated with a one-time token or param would not have the same response body.

I tested intercepting `requestfinished`, too and encountered the same error in Firefox. Again, this does not happen with Chromium.

I realise this may be difficult to reproduce without me providing the API in question, I will try to find a public API that gives the same error.

**Comment by mxschmitt** [comment]
> I realise this may be difficult to reproduce without me providing the API in question, I will try to find a public API that gives the same error.

Exactly, without a full reproducible we are unfortunately not able to act on it.

So far this sounds like unexpected behaviour.

**Comment by michaeleveringham** [comment]
I tried to test this with a local Flask app that produces pseudo random numbers. The idea being that I could see if `response.body()` differed on repeat calls, however, the playwirght instance crashes. Tried on Windows and I encounter an error followed by an immediate crash: `[WinError 10038] an operation was attempted on something that is not a socket`. 

Same behaviour with Playwright 1.33.0.

Flask app:
```
import random
from flask import Flask

app = Flask(__name__)
@app.route("/num/", methods=["GET"])
def num():
    return str(random.random())
app.run(host="127.0.0.1", port="8080")

```

Same playwright example as above without a proxy and with `goto` url of `"http://127.0.0.1:8080/num/"`. I'll keep looking for a public site that mimics this issue.

**Comment by mxschmitt** [comment]
Friendly ping, without a full reproducible, we cannot act on it and would have to close it unfortunately. Thanks for your understanding.

**Comment by michaeleveringham** [comment]
Unfortunately I cannot find a good example. I'll close this out, thanks!
